src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,316 @@
+                               "\n Valid options are [none, mean, sum]");
+            }
+        }
+
+        // ignore_index is optional attribute, assign this as a scalar literal input to the op
+        auto ignore_index =
+            info.add_literal(migraphx::literal(migraphx::shape(shape::int64_type, {1}, {0}), {-1}));
+        bool has_ignore_index = contains(info.attributes, "ignore_index");
+        if(has_ignore_index)
+        {
+            auto ignore_index_val =
+                parser.parse_value(info.attributes.at("ignore_index")).at<int64_t>();
+            ignore_index = info.add_literal(migraphx::literal(
+                migraphx::shape(migraphx::shape::int64_type, {1}, {0}), {ignore_index_val}));
+        }
+
+        // Get and validate Inputs
+        auto scores = args.at(0);
+        auto labels = args.at(1);
+        auto scores_shape = scores->get_shape();
+        auto label_shape = labels->get_shape();
+
+        if(scores_shape.ndim() < 2)
+        {
+            MIGRAPHX_THROW("softmaxcrossentropyloss: Scores must be two or more dimensions [batch, "
+                           "class_size, D1...Dk]");
+        }
+
+        std::set<migraphx::shape::type_t> supported_label_types = {migraphx::shape::int32_type,
+                                                                   migraphx::shape::int64_type};
+        if(not contains(supported_label_types, label_shape.type()))
+        {
+            MIGRAPHX_THROW("softmaxcrossentropyloss: Label must be either int32 or int64 type");
+        }
+
+        // TODO: Update this operator when we get bfloat16 support
+        std::set<migraphx::shape::type_t> supported_score_weight_types = {
+            migraphx::shape::half_type, migraphx::shape::float_type, migraphx::shape::double_type};
+        if(not contains(supported_score_weight_types, scores_shape.type()))
+        {
+            MIGRAPHX_THROW(
+                "softmaxcrossentropyloss: Score must be either half, float, or double type");
+        }
+
+        if(scores_shape.lens()[0] != label_shape.lens()[0])
+        {
+            MIGRAPHX_THROW(
+                "softmaxcrossentropyloss: Score and Labels must identical batch size inputs");
+        }
+
+        if((scores_shape.ndim() - 1) != label_shape.ndim())
+        {
+            MIGRAPHX_THROW(
+                "softmaxcrossentropyloss: Score and Labels must contain identical K-Dimensions");
+        }
+
+        bool is_k_dim  = (scores_shape.ndim() > 3);
+        size_t batch_size = scores_shape.lens().at(0);
+        size_t class_size = scores_shape.lens().at(1);
+
+        auto loss_tensor = info.add_instruction(
+            migraphx::make_op("convert", {{"target_type", scores_shape.type()}}), labels);
+
+        // Default weights will always be 1
+        std::vector<float> weight_vec(class_size, 1.0f);
+        auto weights = info.add_literal(
+            migraphx::literal(migraphx::shape(scores_shape.type(), {class_size}), weight_vec));
```

Comment:
I think it would be better to insert a scalar 1 and broadcast it to the tensor size.

Reply:
All good I can change this. You're thinking broadcast + next ops can be fused easier than with a large literal?

Reply:
Done btw